elide-contrib/elide-test-helpers/src/main/java/com/yahoo/elide/contrib/testhelpers/graphql/GraphQLDSL.java

```diff
@@ -0,0 +1,424 @@
+     * @return a field that represents an non-reltionship entity attribute
+     *
+     * @see <a href="https://graphql.org/learn/schema/#scalar-types">Scalar Field</a>
+     */
+    public static Selection field(String name) {
+        return new ScalarField(name);
+    }
+
+    /**
+     * Creates a complete set of {@link Argument}s that is passed to a object field selection.
+     *
+     * @param arguments  A variable-sized group of {@link Argument}s passed to the object field selection
+     *
+     * @return a complete specification of selection arguments
+     *
+     * @see <a href="https://graphql.org/learn/queries/#arguments">Arguments</a>
+     */
+    public static Arguments arguments(Argument... arguments) {
+        return new Arguments(Arrays.asList(arguments));
+    }
+
+    /**
+     * Creates a {@link Argument single argument} that is passed to a object field selection.
+     *
+     * @param argument  The single {@link Argument} passed to the object field selection
+     *
+     * @return a complete specification of selection argument
+     *
+     * @see <a href="https://graphql.org/learn/queries/#arguments">Arguments</a>
+     */
+    public static Arguments argument(Argument argument) {
+        return new Arguments(Collections.singletonList(argument));
+    }
+
+    /**
+     * Creates a spec object of an GraphQL {@link Argument argument}.
+     *
+     * @param name  The name of the argument
+     * @param value  An object that models argument value.
+     * @return
+     *
+     * @see <a href="https://graphql.org/learn/queries/#arguments">Arguments</a>
+     */
+    public static Argument argument(String name, ValueWithVariable value) {
+        return new Argument(name, value);
+    }
+
+    /**
+     * Creates a string value that is quoted in its serialized form
+     *
+     * @param value  The chars surrounded in a pair of double-quotes
+     *
+     * @return a GraphQL spec
+     */
+    public static ValueWithVariable stringValue(String value) {
+        return new StringValue(value);
+    }
+
+    /**
+     * Creates a query spec that represents a variable, whose variable name follows a {@code $} sign.
+     * <p>
+     * Foe example, {@code $episode}. In this case "episode" is the variable name
+     *
+     * @param name  A GraphQL variable literal without {@code $} sign
+     *
+     * @return a GraphQL variable spec
+     *
+     * @see <a href="https://graphql.org/learn/queries/#variables">Variables</a>
+     */
+    public static ValueWithVariable variableValue(String name) {
+        return new VariableValue(name);
+    }
+
+    /**
+     * Creates a query spec that represents a object value.
+     * <p>
+     * For example, "id:\"1\", title:\"update title\"" as in "{ id:\"1\", title:\"update title\" }".
+     *
+     * @param object  A string representation of the object wrapped
+     *
+     * @return a GraphQL spec
+     */
+    public static ValueWithVariable objectValueWithVariable(String object) {
```

Comment:
I wonder if we could use Jackson and lombok builders here to serialize something like:

```java
Book.builder()
   .id("123")
   .title("xyz")
   .build()
```

If we have to test very complex object graph insertion cases, it will be hard to do that with just a String.

Reply:
Wow why didn't I came up with this great idea?